tests/unit/changeset-test.js

```diff
@@ -830,4 +838,166 @@ test('content can be an empty hash', function(assert) {
+  dummyChangeset.set('org.usa.ma', { name: 'Massachusetts' });
+  dummyChangeset.execute();
+  assert.deepEqual(get(dummyChangeset, 'change'), expectedResult, 'should have correct shape');
+  assert.deepEqual(get(dummyModel, 'org'), expectedResult.org, 'should set value');
+  assert.deepEqual(get(dummyModel, 'org.asia'), expectedResult.org.asia, 'should set value');
+  assert.deepEqual(get(dummyModel, 'org.usa'), expectedResult.org.usa, 'should set value');
+  assert.deepEqual(get(dummyModel, 'org.usa.ca'), expectedResult.org.usa.ca, 'should set value');
+  assert.deepEqual(get(dummyModel, 'org.usa.ny'), expectedResult.org.usa.ny, 'should set value');
+  assert.deepEqual(get(dummyModel, 'org.usa.ma'), expectedResult.org.usa.ma, 'should set value');
+  assert.deepEqual(get(dummyModel, 'org.usa.ma.name'), expectedResult.org.usa.ma.name, 'should set value');
+  expectedResult.org.usa.or = 'or';
+  dummyChangeset.rollback();
+  dummyChangeset.set('org.usa.or', 'or');
+  dummyChangeset.execute();
+  assert.deepEqual(get(dummyModel, 'org.usa.or'), expectedResult.org.usa.or, 'should reset value');
+  assert.deepEqual(get(dummyModel, 'org.usa.ny'), expectedResult.org.usa.ny, 'should reset value');
+  assert.deepEqual(get(dummyModel, 'org.asia'), expectedResult.org.asia, 'should reset value');
+});
+
+test('can update nested keys after rollback changes.', function(assert) {
+  let expectedResult = {
+    org: {
+      asia: { sg: 'sg' },
+      usa: {
+        ny: 'ny',
+        ma: { name: 'Massachusetts' }
+      }
+    }
+  };
+  set(dummyModel, 'org', {
+    asia: { sg: null },
+    usa: {
+      ny: null,
+      ma: { name: null }
+    }
+  });
+  let dummyChangeset = new Changeset(dummyModel, dummyValidator);
+  dummyChangeset.set('org.asia.sg', 'sg');
+  dummyChangeset.set('org.usa.ny', 'ny');
+  dummyChangeset.set('org.usa.ma', { name: 'Massachusetts' });
+  dummyChangeset.execute();
+  assert.deepEqual(get(dummyModel, 'org'), expectedResult.org, 'should set value');
+  assert.deepEqual(get(dummyModel, 'org.usa.ny'), expectedResult.org.usa.ny, 'should set value');
+  assert.deepEqual(get(dummyModel, 'org.asia'), expectedResult.org.asia, 'should have updated value');
+  assert.deepEqual(get(dummyModel, 'org.usa.ma.name'), expectedResult.org.usa.ma.name, 'should set value');
+  expectedResult.org.usa.or = 'or';
+  dummyChangeset.rollback();
+  dummyChangeset.set('org.usa.or', 'or');
+  dummyChangeset.execute();
+  assert.deepEqual(get(dummyModel, 'org'), expectedResult.org, 'should set value');
+  assert.deepEqual(get(dummyModel, 'org.usa.or'), expectedResult.org.usa.or, 'should have updated value');
+  assert.deepEqual(get(dummyModel, 'org.usa.ny'), expectedResult.org.usa.ny, 'should have updated value');
+  assert.deepEqual(get(dummyModel, 'org.asia'), expectedResult.org.asia, 'should have updated value');
+  assert.deepEqual(get(dummyModel, 'org.usa.ma.name'), expectedResult.org.usa.ma.name, 'should set value');
+});
```

Comment:
This test also seems really similar to the last test. Perhaps we can remove this one?